src/coord/sink_connector.rs

```diff
@@ -72,3 +74,34 @@ async fn build_kafka(
         url: builder.broker_url,
     }))
 }
+
+fn build_avro_ocf(
+    builder: AvroOcfSinkConnectorBuilder,
+    id: GlobalId,
+) -> Result<SinkConnector, failure::Error> {
+    let mut path = builder.path.clone();
+
+    let file_name = match (path.file_stem(), path.extension()) {
+        (Some(stem), Some(extension)) => format!(
+            "{}-{}-{}.{}",
+            stem.to_str()
+                .ok_or_else(|| format_err!("error converting file name to utf8 string"))?,
+            id,
+            &builder.file_name_suffix,
+            extension
+                .to_str()
+                .ok_or_else(|| format_err!("error converting file extension to utf8 string"))?
+        ),
+        (Some(stem), None) => format!(
+            "{}-{}-{}",
+            stem.to_str()
+                .ok_or_else(|| format_err!("error converting file name to utf8 string"))?,
+            id,
+            &builder.file_name_suffix,
+        ),
+        (None, _) => bail!("unable to read file name from path {:?}", path),
+    };
+
+    path.set_file_name(file_name);
+    Ok(SinkConnector::AvroOcf(AvroOcfSinkConnector { path }))
```

Comment:
With a bit of care we can avoid requiring UTF-8 paths:

```rust
    let mut name = match builder.path.file_stem() {
        None => bail!(
            "unable to read file name from path {}",
            builder.path.display()
        ),
        Some(stem) => stem.to_owned(),
    };
    name.push("-");
    name.push(id.to_string());
    name.push("-");
    name.push(builder.file_name_suffix);
    if let Some(extension) = builder.path.extension() {
        name.push(".");
        name.push(extension);
    }
    Ok(SinkConnector::AvroOcf(AvroOcfSinkConnector {
        path: builder.path.with_file_name(name),
    }))
```

Reply:
(warning: untested)

Reply:
this was a great suggestion that looks like it worked